### DISCUSSION TITLE
Lack of USB Mass Storage support for RP2 port - have I missed a critical difference moving from STM32?

### DISCUSSION THREAD

**Discussion by DoubleVee73** [micropython/micropython#10926] [RP2040 / Pico]
I think I may have messed up badly. 

I have a product in advanced stages of development (based upon the STM32F411) but have been hit with the chip shortage and now huge price hikes. So I have moved the design to the RP2040 (and I wanted to support British), have tested everything in prototype builds but I have missed a critical feature.

The product requires a customer supplied config file. A button press during power on was used to put the STM32 into USB mass storage mode to expose the flash, config.json was copied across from any connected laptop and the product restarted. However, I am not able to access the file system on the RP2 port (as I currently understand things). Am I correct? Have I missed a rather critical difference here for my application?

At a stretch, I could get customers to install something 'light' (like mpremote but it would need to be a standalone app), and use this to copy the file to the product, but I am reaching out to the community to see if there is a better solution or recommendation at this stage. The product will use a frozen build so the only file visible to a customer will be config.json.

Your advice and any pointers at this stage would be so welcome - thank you.

**Comment by robert-hh** [comment]
If you build your software yourself, you can try to enable MSC support by adding the line:
```
// Enable USB Mass Storage with FatFS filesystem.
#define MICROPY_HW_USB_MSC              (1)
```
to mpconfigboard.h.

**Reply by robert-hh** [reply]
However it will be available permanently. There were some discussion to make it configurable by the python script, but that is not settled.

**Reply by DoubleVee73** [reply]
Thank you Robert for replying - if I could find a way to toggle the feature that could work but I would have no idea where to start. I will do some reading for other features to see if a mechanism exists.

I am also concerned now about simple USB serial comms. Again, on the STM32 I used the usb_vcp to communicate with a controller module I am also building. I cannot see any reference to this feature on the RP2 port!

Maybe I have jumped the gun moving to the RP2040 as there is basic USB stuff still missing that no doubt will be addressed in the next year or two

**Reply by martin-ringaly** [reply]
Hi @DoubleVee73, did you find a way to programmatically enable USB MSC mode?

**Reply by Josverl** [reply]
@martin-ringaly , this might be worth following on Discord : https://discord.com/channels/574275045187125269/1012726673709469818/1411253746725294091

**Reply by scruss** [reply]
that discord link is giving me the infinite wait. Maybe it's a private group?

**Reply by Josverl** [reply]
MP discord > MicroPython-dev channel

**Reply by scruss** [reply]
does a weird re-login thing, but gets there eventually. This is why Discord is not a place of record

**Comment by sophiedegran** [comment]
I'm thinking about other simple thing..
I don't know if it's possible for you..
Installing microdot with a simple upload page where you can 'upload' your config file..
That's what I do for certain of my project to modify main.py..
The avantage is no usb manipulation and you can verify 'checksum' file before acting...
